File: Arrays & Hashing/128. Longest Consecutive Sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        res = 0
        num_set = set(nums)

        for i in num_set:

            if (i - 1) in num_set:
                continue
            else:
                length = 1
                current_num = i

                while (current_num + 1) in num_set:
                    length += 1
                    current_num = current_num + 1

                if length > res:
                    res = length

        return res

File: C-Sharp Solutions/Arrays & Hashing/128. Longest Consecutive Sequence.cs
public class Solution {
    public int LongestConsecutive(int[] nums) {
        HashSet<int> num_set = new HashSet<int>(nums);
        int longest_seq = 0;

        foreach (int num in num_set) {
            if (num_set.Contains(num - 1)) {
                continue;
            } else {
                int temp_num = num;
                int current_len = 0;

                while (num_set.Contains(temp_num)) {
                    current_len++;
                    temp_num++;
                }

                if (current_len > longest_seq) {
                    longest_seq = current_len;
                }
            }
        }

        return longest_seq;

    }
}
